tasks/src/main/java/ru/sberbank/school/task02/calculator/ExtendedCurrencyCalculator.java

```diff
@@ -0,0 +1,95 @@
+package ru.sberbank.school.task02.calculator;
+
+import ru.sberbank.school.task02.ExtendedFxConversionService;
+import ru.sberbank.school.task02.ExternalQuotesService;
+import ru.sberbank.school.task02.util.Beneficiary;
+import ru.sberbank.school.task02.util.ClientOperation;
+import ru.sberbank.school.task02.util.Quote;
+import ru.sberbank.school.task02.util.Symbol;
+import java.math.BigDecimal;
+import java.util.*;
+
+public class ExtendedCurrencyCalculator extends CurrencyCalculator implements ExtendedFxConversionService {
+    private final ExternalQuotesService externalQuotesService;
+
+    public ExtendedCurrencyCalculator(ExternalQuotesService externalQuotesService) {
+        super(externalQuotesService);
+        this.externalQuotesService = externalQuotesService;
+    }
+
+    private Optional<BigDecimal> extendedOperation(ClientOperation operation,
+                                                   Beneficiary beneficiary,
+                                                   List<QuotePrice> quotePrices,
+                                                   BigDecimal amountOfRequest) {
+
+        Optional<QuotePrice> quotePrice = findQuote(new CompareQutesExtend(),
+                quotePrices,
+                amountOfRequest, new CompareQuotesBeneficiary(operation, beneficiary));
+
+        if (!quotePrice.isPresent()) {
+            return Optional.empty();
+        }
+        return Optional.of(quotePrice.get().getPricePerPiece());
+    }
+
+    Optional<QuotePrice> findQuote(Comparator<QuotePrice> compareByAmount,
+                              List<QuotePrice> quotePrices,
+                              BigDecimal amountOfRequest,
+                              Comparator<QuotePrice> compareByBeneficiary) {
+
+        List<QuotePrice> finalQuoteList = new ArrayList<>();
+        quotePrices.sort(compareByAmount);
+        BigDecimal lowerLimit = BigDecimal.ZERO;
+
+        for (QuotePrice quotePrice: quotePrices) {
+            quotePrice.setLowerLimit(lowerLimit);
+            if (quotePrice.checkQuoteInMyScope(amountOfRequest)) {
+               finalQuoteList.add(quotePrice);
```

Comment:
&#39;if&#39; child has incorrect indentation level 15, expected level should be 16.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a2067479262>*